src/arrayMethodReduce.js

```diff
@@ -6,6 +6,25 @@
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
     // write code here
+    if (this.length === 0) {
+      return initialValue;
+    }
+
+    let result;
+    let startIndex = 0;
+
+    if (arguments.length > 1) {
+      result = initialValue;
+    } else {
+      result = this[0];
+      startIndex++;
+    }
```

Comment:
Можно изначально при объявлении присвоить переменной значение для одного случая, а после для другого случая используя if изменить это значение
```suggestion
    let result = this[0];
    let startIndex = 1;

    if (arguments.length > 1) {
      result = initialValue;
      startIndex = 0;
   }
```
ну или наоборот